tests/test_datatypes.py

```diff
@@ -102,6 +98,9 @@ def test_mysql_datatypes(self) -> None:
             # Same type and length
             coldata.col("string", "VARCHAR(128)", length=128)
```

Comment:
I'm not entirely sure that I understand the logic that this and the earlier CHAR/VARCHAR/NVARCHAR clauses are testing for objects with lengths.

Is this it?

> A type with a Felis `length` has a redundant override either when the override type matches and is stated without an explicit size (i.e., "VARCHAR" for length-8 "string") or when the override type matches and has an explicit length matching the Felis `length` (i.e. "CHAR(10)" for length-10 "char").

Reply:
> I'm not entirely sure that I understand the logic that this and the earlier CHAR/VARCHAR/NVARCHAR clauses are testing for objects with lengths.
> 
> Is this it?
> 
> > A type with a Felis `length` has a redundant override either when the override type matches and is stated without an explicit size (i.e., "VARCHAR" for length-8 "string") or when the override type matches and has an explicit length matching the Felis `length` (i.e. "CHAR(10)" for length-10 "char").

Yes, a redundant override of this type would look like this in YAML

```
datatype: char
length: 12
mysql:datatype: CHAR(12)
```

These compile to the same SQL of `CHAR(12)` and so they are considered the same.

Does that answer your question?